ecs-cli/modules/config/readwriter_v1.go

```diff
@@ -0,0 +1,155 @@
+// compose-service-name-prefix:
+// cfn-stack-name-prefix: ecs-cli-
+type YAMLReadWriter struct {
+	destination *Destination
+}
+
+// NewReadWriter creates a new Parser object.
+func NewReadWriter() (*YAMLReadWriter, error) {
+	dest, err := newDefaultDestination()
+	if err != nil {
+		return nil, err
+	}
+
+	return &YAMLReadWriter{destination: dest}, nil
+}
+
+func readYAML(yamlPath string, configMap map[interface{}]interface{}, cliConfig *CLIConfig) error {
+	// convert yaml to CliConfig
+	dat, err := ioutil.ReadFile(yamlPath)
+	if err != nil {
+		return err
+	}
+	if err = yaml.Unmarshal(dat, cliConfig); err != nil {
+		return err
+	}
+
+	// convert yaml to a map (replaces IsKeyPresent functionality)
+	if err = yaml.Unmarshal(dat, &configMap); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// GetConfig gets the ecs-cli config object from the config file.
+// map contains the keys that are present in the config file (maps string field name to string field value)
+// map is type map[interface{}]interface{} to ensure fowards compatibility with changes that will
+// cause certain keys to be mapped to maps of keys
+func (rdwr *YAMLReadWriter) GetConfig() (*CLIConfig, map[interface{}]interface{}, error) {
+	to := new(CLIConfig)
+	configMap := make(map[interface{}]interface{})
+	// read the raw bytes of the config file
+	iniPath := iniConfigPath(rdwr.destination)
+	yamlPath := yamlConfigPath(rdwr.destination)
+
+	_, iniErr := os.Stat(iniPath)
+	_, yamlErr := os.Stat(yamlPath)
+	if yamlErr == nil {
+		if err := readYAML(yamlPath, configMap, to); err != nil {
+			return nil, nil, err
+		}
+	} else if iniErr == nil { // file exists
+		// old ini config
+		iniReadWriter, err := NewINIReadWriter(rdwr.destination)
+		if err != nil {
+			return nil, nil, err
+		}
+		to, configMap, err = iniReadWriter.GetConfig()
+		if err != nil {
+			return nil, nil, err
+		}
+
+	} else {
+		// if neither file existed we return the yaml error
+		return nil, nil, yamlErr
+	}
+	return to, configMap, nil
+}
+
+// Save saves the CLIConfig to a yaml formatted file
+func (rdwr *YAMLReadWriter) Save(cliConfig *CLIConfig) error {
+	destMode := rdwr.destination.Mode
+	if err := os.MkdirAll(rdwr.destination.Path, *destMode); err != nil {
+		return err
+	}
+	// set version
+	cliConfig.Version = configVersion
+
+	path := yamlConfigPath(rdwr.destination)
+
+	// Warn the user if ini path also exists
+	iniPath := iniConfigPath(rdwr.destination)
+	if _, iniErr := os.Stat(iniPath); iniErr == nil {
+		logrus.Warnf("Writing yaml formatted config to %s/.ecs/%s.\nIni formatted config still exists in %s/.ecs/%s.", os.Getenv("HOME"), yamlConfigFileName, os.Getenv("HOME"), iniConfigFileName)
+	}
+
+	// If config file exists, set permissions first, because we may be writing creds.
+	if _, err := os.Stat(path); err == nil {
+		if err = os.Chmod(path, configFileMode); err != nil {
+			logrus.Errorf("Unable to chmod %s to mode %s", path, configFileMode)
+			logrus.Error(err.Error())
```

Comment:
You can combine this with the previous error line itself. Also, with `logrus`, you're discouraged to use `Errorf`m but instead do `WithFields`. Please replace all `Errorf`, `Info` etc with `WithFields` method.